models/yiiModels/DataAnalysisAppSearch.php

```diff
@@ -0,0 +1,232 @@
+<?php
+
+//******************************************************************************
+//                         DataAnalysisAppSearch.php
+// SILEX-PHIS
+// Copyright © INRA 2019
+// Creation date: 25 Feb, 2019
+// Contact: arnaud.charleroy@inra.fr, anne.tireau@inra.fr, pascal.neveu@inra.fr
+//******************************************************************************
+
+namespace app\models\yiiModels;
+
+use Yii;
+use openSILEX\opencpuClientPHP\OpenCPUServer;
+use app\models\wsModels\WSConstants;
+use yii\helpers\Url;
+use Symfony\Component\Yaml\Yaml;
+use GuzzleHttp\Exception\ClientException;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7;
+use openSILEX\opencpuClientPHP\classes\CallStatus;
+
+include_once '../config/web_services.php';
+require_once '../config/config.php';
+
+/**
+ * DataAnalysisAppSearch search class which makes link
+ * between openSILEX and OpenCPU
+ * @author Arnaud Charleroy <arnaud.charleroy@inra.fr>
+ */
+class DataAnalysisAppSearch {
+    /**
+     *
+     * @var OpenCPUServer 
+     */
+    public $ocpuserver;
+
+    /**
+     * Configuration file constants
+     */
+    const VIGNETTE_IMAGE = "vignette";
+    const APP_SHORT_NAME = "appShortName";
+    const R_PACKAGE_NAME = "packageName";
+    const APP_INDEX_HREF = "appIndexHref";
+    const FUNCTION_NAME = "functionName";
+    const FUNCTION_HELP = "functionHelp";
+    const AVAILABLE_FUNCTIONS = "availableFunctions";
+    const FUNCTION_DESCRIPTION = "description";
+    const INTEGRATED_FUNCTION = "integratedFunctions";
+    
+    const DEFAULT_DEMO_APP = "opensilex/opensilex-datavis-rapp-demo";
+
+    /**
+     * Initialize openCPU server connection
+     * @param type $verbose if true, give connection metrics information
+     */
+    public function __construct($verbose = false) {
+        $this->ocpuserver = new OpenCPUServer(\config::path()['ocpuServer']);
+        if ($verbose) {
+            $this->ocpuserver::$ENABLE_CALL_STATS = true;
+        }
+    }
+
+    /**
+     * Remove a choosen app (demo app)
+     * @param array $appList a list of app (github repo list)
+     * @return array list without self::DEFAULT_DEMO_APP app
+     */
+    private function removeDemoAppFromAppList($appList) {
+        foreach ($appList as $key => $app) {
+            if ( $app == self::DEFAULT_DEMO_APP) {
+                unset($appList[$key]) ;
+            }
+        }
+        return $appList;
+    }
+    
+    /**
+     * List all available apps
+     * @param array $sessionToken used for the data access
+     * @param string $params search params
+     * @return array list of app with their informations
```

Comment:
Add an example of the returned array.

Reply:
Add more explaination of the process and link between app and opencpu, add example